Tests/PrettyPrinterTests/EnumDeclTests.swift

```diff
@@ -0,0 +1,301 @@
+public class EnumDeclTests: PrettyPrintTestCase {
+  public func testBasicEnumDeclarations() {
+    let input =
+      """
+      enum MyEnum {
+        case firstCase
+        case secondCase
+      }
+      public enum MyEnum {
+        case firstCase
+        case secondCase
+      }
+      public enum MyLongerEnum {
+        case firstCase
+        case secondCase
+      }
+      """
+
+    let expected =
+      """
+      enum MyEnum {
+        case firstCase
+        case secondCase
+      }
+      public enum MyEnum {
+        case firstCase
+        case secondCase
+      }
+      public enum
+      MyLongerEnum {
```

Comment:
The style guide doesn't really address this (because why would someone have a type name long enough to cause this), but my gut feeling is that this should wrap like this:

```swift
public enum
  MyLongerEnum
{
  case firstCase
  case secondCase
}
```

How horrible would that make things?

Reply:
In isolation, yes, but I wonder how it interacts with other constructs:

```swift
public enum
  MyLongerEnum<
    Foo,
    Bar
  >:
    SomeProtocol,
    OtherProtocol
{
```

I'm kind of less convinced now.

Reply:
Chatted off-line. I formally retract this suggestion after thinking about it a bit more. 🙂